### PR TITLE
fix emission prices in models with non-equidistant period lengths (#167)

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Description of the PR

This PR fixes an issue where the auxiliary reporting variable `PRICE_EMISSION` was not correctly computed when the period durations where not equal across the model horizon. The PR also adds a number of unit tests to verify that emissions prices are calculated as expected in a number of stylized cases.

related to https://github.com/iiasa/message_ix/issues/167, will be closed when merging `message_ix_v1.2` into `master`

### CODE REVIEW

File: tests/test_feature_price_emission.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
+from message_ix import Scenario
```
Comment (khaeru) [inline tests/test_feature_price_emission.py]: In this file:
- add a docstring for `model_setup()`
- add a plain language comment stating the expected behaviour for each assertion. I'll add an example below.

File: tests/test_feature_price_emission.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+
+def test_no_constraint(test_mp):
+    scen = Scenario(test_mp, MODEL, 'no_constraint', version='new')
+    model_setup(scen, [2020, 2030])
+    scen.commit('initialize test scenario')
+    scen.solve()
+
+    assert scen.var('OBJ')['lvl'] == 0
+    assert scen.var('PRICE_EMISSION').empty
```
Comment (khaeru) [inline tests/test_feature_price_emission.py]: E.g.
```
# Variable PRICE_EMISSION has no value when emissions are not constrained
assert scen.var('PRICE_EMISSION').empty
```

**Comment by danielhuppmann** [comment]
@khaeru, thanks for the comments - all done!

**Comment by khaeru** [comment]
Added comments in tests make them quite easily intelligible—a great example to set for other developers!

**Comment by danielhuppmann** [comment]
merging into the release-candidate-staging branch to continue working on the `ixmp` and `ixmp_source` dependencies

**Pull request merged**